packages/user-ui/src/pages/project/controller.rs

```diff
@@ -22,253 +22,71 @@ use crate::{
-                    attributes: vec![],
-                    org_id: 1,
-                }],
-                panel_counts: vec![PanelCountsV2 {
-                    created_at: 1741103145,
-                    updated_at: 1741103145,
-                    panel_id: 1,
-                    panel_survey_id: 1,
-                    user_count: 100,
-                }],
-                noncelab_id: Some(1),
-                response_count: 50,
-            }],
-            members: vec![
-                DeliberationUser {
-                    id: 1,
-                    created_at: 1741103145,
-                    updated_at: 1741103145,
-                    user_id: 1,
-                    deliveration_id: 1,
-                    role: models::Role::Analyst,
-                },
-                DeliberationUser {
-                    id: 2,
-                    created_at: 1741103145,
-                    updated_at: 1741103145,
-                    user_id: 2,
-                    deliveration_id: 1,
-                    role: models::Role::Admin,
-                },
-                DeliberationUser {
-                    id: 3,
-                    created_at: 1741103145,
-                    updated_at: 1741103145,
-                    user_id: 3,
-                    deliveration_id: 1,
-                    role: models::Role::DeliberationAdmin,
-                },
-            ],
-            panels: vec![PanelV2 {
-                id: 1,
-                created_at: 1741103145,
-                updated_at: 1741103145,
-                name: "패널".to_string(),
-                user_count: 100,
-                attributes: vec![],
-                org_id: 1,
-            }],
-            comments: vec![],
-            response_count: 10,
-        }
-    }
-}
-
 #[derive(Clone, Debug, Default, PartialEq)]
 pub struct SurveyResponses {
     pub answers: IndexMap<i64, (String, ParsedQuestion)>, // question_id, (title, response_count, <panel_id, answer>)
 }
 
 #[derive(Debug, Clone, Copy)]
-pub struct SampleController {
+pub struct Controller {
     answers: Signal<Vec<Answer>>,
     // NOTE: Whether I have ever filled out a survey
     // NOTE: In the future, it will be linked to the API and the relevant part should be checked.
     check_edit: Signal<bool>,
     pub survey_responses: Signal<SurveyResponses>,
 }
 
-impl SampleController {
+impl Controller {
     pub fn init(lang: Language) -> std::result::Result<Self, RenderError> {
         let mut ctrl = Self {
             answers: use_signal(|| vec![]),
             check_edit: use_signal(|| false),
             survey_responses: use_signal(|| SurveyResponses::default()),
         };
 
-        use_context_provider(|| ctrl.clone());
+        use_context_provider(|| ctrl);
 
         let questions = ctrl.clone().get_deliberation().surveys[0].questions.clone();
         let responses = ctrl.clone().get_deliberation_responses();
 
-        use_effect({
+        let memoized_answers = use_memo({
```

Comment:
What types of questions and responses?
I think they are not Signal, right?
Does this memo work properly? it seems not to cover any Signal.

Reply:
First of all, I confirmed that it works properly on the chart.
I will check if the relevant part works properly while linking the data and it seems like it needs to be fixed.
I will add a FIXME to indicate that checking is necessary.